### ISSUE TITLE
Validate 3.3.0 erroneously reports data objects out of offset order

### ISSUE THREAD

**Issue by lylehuber** [NASA-PDS-Incubator/validate#754]
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

Used 3.3.0 on the bundle located at
https://atmos.nmsu.edu/PDS/data/PDS4/co_iss_global-maps/

Validation report attached here.
[coissval.txt](https://github.com/NASA-PDS/validate/files/13284711/coissval.txt)

### 🕵️ Expected behavior

I expected to see the schema WARNINGs. I did not expect to see the out of order WARNINGs. Upon inspection, the problem is with Validate not understanding when <File_Area_Obsevrational> ended and when <File_Area_Observational_Supplemental> began. 

### 📜 To Reproduce

```
validate --rule pds4.bundle --target /export/atmos7/PDS/tmp/ROSES/co_iss_global-maps/
```

### 🖥 Environment Info

Linux

### 📚 Version of Software Used

3.3.0

### 🩺 Test Data / Additional context

See above

### 🦄 Related requirements

🦄 #683

### I&T
TestRail Test ID: T8681182

**Comment by tbarnes4** [comment]
We are getting the same issue with our DART archive whether validating as a bundle, collection, or individual label.

**Comment by al-niessner** [comment]
@jordanpadams 

Need clarification on desirements here. The current code lumps all file areas and supplemental into a giant list then sorts them. In this case the file area header and then supplemental text stream both have an offset of 0 but they are different files. Should we break it out by filename or treat each file area as atomic (do not sort them together)? Both fix this error, but are somewhat different in theory. The filename is more robust if different file areas point at the same file. Splitting out the different areas only works if they are truly atomic. What is your preferred poison?

**Comment by al-niessner** [comment]
@jordanpadams 

Never mind. The information available at the time when the offset is tested answered it for you.

**Comment by miguelp1986** [comment]
TestRail Test ID: T8681182
